rasa/cli/data.py

```diff
@@ -234,6 +273,422 @@ def split_nlu_data(args: argparse.Namespace) -> None:
+            output_directory_diverse, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_diverse,
+            changed_intents=changed_intents_diverse,
+            metric=metric,
+            output_file=output_file_diverse,
+        )
+
+        output_file_random = os.path.join(
+            output_directory_random, f"{metric}_changes.png"
+        )
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary_random,
+            changed_intents=changed_intents_random,
+            metric=metric,
+            output_file=output_file_random,
+        )
+
+
+def suggest_nlu_data(args: argparse.Namespace) -> None:
+    """Load NLU training & evaluation data, paraphrases, an existing classification report, its corresponding config
+    file and suggest additional training
+     examples.
+
+    Args:
+        args: Commandline arguments
+    """
+    nlu_training_data = rasa.shared.nlu.training_data.loading.load_data(
+        args.nlu_training_data
+    )
+    nlu_evaluation_data = rasa.shared.nlu.training_data.loading.load_data(
+        args.nlu_evaluation_data
+    )
+    paraphrases = rasa.shared.nlu.training_data.loading.load_data(args.paraphrases)
+    classification_report = rasa.shared.utils.io.read_json_file(
+        args.nlu_classification_report
+    )
+    random.seed(args.random_seed)
+
+    # Determine low data, low performing and frequently confused intents
+    num_intents = len(nlu_training_data.intents)
+    avg_size = int(
+        reduce(
+            lambda acc, num: acc + (num / num_intents),
+            nlu_training_data.number_of_examples_per_intent.values(),
+            0,
+        )
+    )
+
+    low_data_intents = sorted(
+        nlu_training_data.number_of_examples_per_intent.items(),
+        key=operator.itemgetter(1),
+    )[: args.num_intents]
+    low_precision_intents = sorted(
+        map(
+            lambda k: (k, classification_report[k]["precision"]),
+            nlu_training_data.intents,
+        ),
+        key=operator.itemgetter(1),
+    )[: args.num_intents]
+    low_recall_intents = sorted(
+        map(
+            lambda k: (k, classification_report[k]["recall"]), nlu_training_data.intents
+        ),
+        key=operator.itemgetter(1),
+    )[: args.num_intents]
+    low_f1_intents = sorted(
+        map(
+            lambda k: (k, classification_report[k]["f1-score"]),
+            nlu_training_data.intents,
+        ),
+        key=operator.itemgetter(1),
+    )[: args.num_intents]
+    freq_confused_intents = sorted(
+        map(
+            lambda k: (k, sum(classification_report[k]["confused_with"].values())),
+            nlu_training_data.intents,
+        ),
+        key=operator.itemgetter(1),
+        reverse=True,
+    )[: args.num_intents]
+
+    pooled_intents = (
+        set(map(lambda tp: tp[0], low_data_intents))
```

Comment:
These could be `{tp[0] for tp in xyz}` each, or even:

`{tp[0] for tp in low_data_intents + low_precision_intents + ...}`